app/js/app.js

```diff
@@ -1,118 +1,137 @@
-     * @return {Array} An array of unique words
-     */
-    tokenize(fileText) {
-      this.text = fileText;
-      return [...new Set(this.text
-                .toLowerCase()        // Converts text to lower case
-                .replace(/[^\w\s]/g, '')  // Removes any non-word character
-                .split(/\s+/)       // Turns it into an array
-                .sort()           // Sorts array
-      )];
-    }
-    /**
-     * Creates the index for documents
-     * @param  {String} fileName
-     * @param  {Array} content
-     * @return {Object} this.allIndices object
-     */
-    createIndex(fileName, content) {
-      const fileIndex = {};
+  constructor() {
+    this.allIndices = {};
+  }
+  /**
+   * Reads the data from the file being uploaded
+   * @param  {File} file - Uploaded file to be read.
+   * @return {void}
+   */
+  readFile(file) {
+    this.file = file;
+    return new Promise((resolve, reject) => {
+      const bookReader = new FileReader();
 
-      content.forEach((objDoc, index) => {
-        for (const key in objDoc) {
+      bookReader.onload = evt => resolve(evt.target.result);
+      bookReader.onerror = (evt) => {
+        reject(`Error reading + ${this.file.name}: ${evt.target.result}`);
+      };
+      bookReader.readAsText(this.file);
+    });
+  }
+  /**
+   * Ensures all the documents in a particular file is valid
+   * @param  {String} fileName
+   * @param  {Object} fileContent
+   * @return {Boolean} isValid -True or false
+   */
+  validateFile(fileName, fileContent) {
+    this.content = fileContent;
+    this.fileName = fileName;
+    let isValid = true;
+    try {
+      const parsed = JSON.parse(JSON.stringify(this.content));
+      isValid = (parsed.length === 0) ? false : // eslint-disable-line
+        (!this.fileName.toLowerCase().match(/\.json$/g)) ? false : isValid;
+      parsed.forEach((key) => {
+        if (typeof key.title !== 'string' || typeof key.text !== 'string') {
+          isValid = false;
+        }
+      });
+    } catch (error) {
+      isValid = false;
+    }
+    return isValid;
+  }
+  /**
+   * Strips out special characters from documents to be indexed
+   * @param  {String} fileText - String from file to be tokenized
+   * @return {Array} An array of unique words
+   */
+  tokenize(fileText) {
+    this.text = fileText;
+    return [...new Set(this.text
+              .toLowerCase()        // Converts text to lower case
+              .replace(/[^\w\s]/g, '')  // Removes any non-word character
+              .split(/\s+/)       // Turns it into an array
+              .sort()           // Sorts array
+    )];
+  }
+  /**
+   * Creates the index for documents
+   * @param  {String} fileName
+   * @param  {Array} content
+   * @return {Object} this.allIndices object
+   */
+  createIndex(fileName, content) {
+    const fileIndex = {};
+    content.forEach((objDoc, index) => {
+      if (typeof objDoc === 'object') {
```

Comment:
Initially eslint threw some error about using For-In without if-else but I think I get it now. I have to check if that key exist and handle errors thrown.